state/pruner.go

```diff
@@ -217,6 +233,61 @@ func (p *Pruner) SetABCIResRetainHeight(height int64) error {
 	return nil
 }
 
+func (p *Pruner) SetTxIndexerRetainHeight(height int64) error {
+	// Ensure that all requests to set retain heights via the application are
+	// serialized.
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+	if height <= 0 {
+		return ErrInvalidHeightValue
+	}
+
+	currentRetainHeight, err := p.txIndexer.GetRetainHeight()
+	if err != nil {
+		if !errors.Is(err, ErrKeyNotFound) {
+			return err
+		}
+		return p.txIndexer.SetRetainHeight(height)
+	}
+	if currentRetainHeight > height {
+		return ErrPrunerCannotLowerRetainHeight
+	}
+	if err := p.txIndexer.SetRetainHeight(height); err != nil {
+		return err
+	}
+
+	p.metrics.PruningServiceTxIndexerRetainHeight.Set(float64(height))
+	return nil
+}
+
+func (p *Pruner) SetBlockIndexerRetainHeight(height int64) error {
```

Comment:
Who is calling this function? I only see two callsite in the tests. If this function is never called, we won't prune any blocks